x-pack/legacy/plugins/siem/public/pages/detection_engine/create_rule/components/schedule_item_form/index.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFieldNumber, EuiFormRow, EuiSelect } from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import {
+  FieldHook,
+  UseField,
+} from '../../../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib';
+import { getFieldValidityAndErrorMessage } from '../../../../../../../../../../src/plugins/es_ui_shared/static/forms/components/helpers';
+
+import * as I18n from './translations';
+
+interface ScheduleItemProps {
+  field: FieldHook;
+  dataTestSubj: string;
+  idAria: string;
+  isDisabled: boolean;
+}
+
+const timeTypeOptions = [
+  { value: 's', text: I18n.SECONDS },
+  { value: 'm', text: I18n.MINUTES },
+  { value: 'h', text: I18n.HOURS },
+];
+
+const StyledEuiFormRow = styled(EuiFormRow)`
+  .euiFormControlLayout {
+    max-width: 200px !important;
+  }
+`;
+
+export const ScheduleItem = ({ dataTestSubj, field, idAria, isDisabled }: ScheduleItemProps) => {
+  const [timeType, setTimeType] = useState('s');
+  const [timeVal, setTimeVal] = useState<string | number>('');
+  const { isInvalid, errorMessage } = getFieldValidityAndErrorMessage(field);
+
+  const onChangeTimeType = useCallback(e => {
+    setTimeType(e.target.value);
+  }, []);
+
+  const onChangeTimeVal = useCallback(e => {
+    const sanitizedValue = parseInt(e.target.value, 10);
+    setTimeVal(isNaN(sanitizedValue) ? '' : sanitizedValue);
+  }, []);
+
+  useEffect(() => {
+    if (!isEmpty(timeVal) && Number(timeVal) >= 0 && field.value !== `${timeVal}${timeType}`) {
+      field.setValue(`${timeVal}${timeType}`);
+    }
+  }, [field.value, timeType, timeVal]);
```

Comment:
I can go with you in more detail about this component, but I do believe I will avoid any infinity loop because of this condition `field.value !== `${timeVal}${timeType}`